### ISSUE TITLE
Licensing is confusing

### ISSUE THREAD

**Issue by kariharju** [xlwings/xlwings#1933]
#### The license statements are somewhat problematic which probably will churn some users away.
```
xlwings (Open Source) includes all files in the xlwings package except
the pro folder, i.e., the xlwings.pro subpackage and is distributed under
the following BSD 3-clause license
```
Stating a different license agreement for a single folder in a repository seems problematic at best. 
What happens if a file meant for the pro -folder is committed in the wrong folder? Is it OSS forever? ..and vice-versa of course. 

As the IP owners, I think you could make one repo that has only open-source code, then make a fork that contains the commercial code and have clearly separate license statements for both.
..just my two cents.

The library looks quite impressive, but I think I already passed this by once due to the license not being clear.

**Comment by fzumstein** [comment]
Hi Kari,

Appreciate your feedback, here are my thoughts:

> Stating a different license agreement for a single folder in a repository seems problematic at best.

It's common practice, actually. Have a look at [pywin32](https://github.com/mhammond/pywin32) (different license for each subproject) or [GitLab](https://gitlab.com/gitlab-org/gitlab/-/blob/master/LICENSE).

> What happens if a file meant for the pro -folder is committed in the wrong folder? Is it OSS forever? 

Then that file in only that version is open source.

> As the IP owners, I think you could make one repo that has only open-source code, then make a fork that contains the commercial code and have clearly separate license statements for both.

If, then it would have to be the other way around: make a fork/mirror with only the OSS part. However, to get it, all you have to do is:

```
git clone https://github.com/xlwings/xlwings.git
rm -rf xlwings/xlwings/pro
```

Also, having two repos might just confuse the hell out of users who don't like to read READMEs (i.e., the majority).

I do think, however, that you have a point and I already had plans to improve things in the following way:

* Improve the README to make it simpler to understand. Something along the lines `Everything in this repo is licensed under BSD 3-clause license except the xlwings/pro folder whose license you can see here. If you want an OSS compatible source, simply delete that folder.`
* Add a command so that pip does not install the `pro` folder. Something like `pip install xlwings[OSS]` Note that even today, you can't accidentally use the pro version if you don't want to as it asks you for a license key.
* Mention `open core` somewhere in the README

**Comment by kariharju** [comment]
Makes sense, and I like the improvement ideas as I think those would really clarify things.

This is a confusion that only people in my kind of role hit: The first thing to check on any repo is the license. 
If there is any confusion either around the OSS license text or "what do I buy and how?" it is usually a quick pass. 
So you may have some churn over this :wink:

I just got a ping from one of our RPA customers on this library yesterday and remembered that I probably had looked into this earlier.. then looked at the license and remembered why...  so that is why I pinged on this.

We'll probably make a demo bot with this library to get more visibility as the customers we are working with are fine with buying these kinds of licenses (some even prefer buying just to avoid the governance around OSS) as long as the "rules are clear".

Anyway thanks for the reply, and for my part, this issue can be closed.

**Comment by fzumstein** [comment]
I'll keep it open until I've implemented the improvements.
